src/cudamatrix/cu-matrix.cc

```diff
@@ -1439,6 +1439,53 @@ void CuMatrixBase<Real>::DiffTanh(const CuMatrixBase<Real> &value,
   }
 }
 
+
+template<typename Real>
+void CuMatrixBase<Real>::ReLU(const CuMatrixBase<Real> &src) {
+  KALDI_ASSERT(SameDim(*this, src));
+#if HAVE_CUDA == 1
+  if (CuDevice::Instantiate().Enabled()) {
+    Timer tim;
+
+    dim3 dimBlock(CU2DBLOCK, CU2DBLOCK);
+    dim3 dimGrid(n_blocks(src.NumCols(), CU2DBLOCK), n_blocks(src.NumRows(), CU2DBLOCK));
+
+    cuda_tanh(dimGrid, dimBlock, this->data_, src.data_, this->Dim(), src.Stride());
```

Comment:
i think you mean relu not tanh here- the same in the derivative code.  Did you test this?
Also, there are already existing low-level functions that would allow you to implement relu, there is no need to change the cudamatrix/ library.  Look at how it's implemented in nnet2 and nnet3.